be/src/runtime/runtime_filter_worker.cpp

```diff
@@ -509,10 +556,63 @@ void RuntimeFilterWorker::_receive_total_runtime_filter(PTransmitRuntimeFilterPa
         }
 
         index += (1 + half);
+        _exec_env->add_rf_event(request.query_id(), request.filter_id(), addr.hostname, "FORWARD");
         send_rpc_runtime_filter(stub, rpc_closure, default_send_rpc_runtime_filter_timeout_ms, request);
     }
 }
 
+void RuntimeFilterWorker::_process_send_broadcast_runtime_filter_event(
+        PTransmitRuntimeFilterParams&& params, std::vector<TRuntimeFilterDestination>&& destinations, int timeout_ms) {
+    DCHECK(!destinations.empty());
+    std::random_device rd;
+    std::mt19937 rand(rd());
+    std::shuffle(destinations.begin(), destinations.end(), rand);
+    _exec_env->add_rf_event(params.query_id(), params.filter_id(),
+                            strings::Substitute("SEND_BROADCAST_RF_RPC: num_dest=$0", destinations.size()));
+    params.set_is_partial(false);
+    TNetworkAddress local;
+    local.hostname = BackendOptions::get_localhost();
+    local.port = config::brpc_port;
+    for (auto i = 1; i < destinations.size(); ++i) {
+        if (destinations[i].address == local) {
+            std::swap(destinations[i], destinations[0]);
+            break;
+        }
+    }
+
+    size_t k = 0;
+    static constexpr size_t NUM_INFLIGHT_RPC = 10;
+    while (k < destinations.size()) {
+        auto num_inflight = std::min(destinations.size() - k, NUM_INFLIGHT_RPC);
+        std::vector<RuntimeFilterRpcClosure*> rpc_closures(num_inflight);
```

Comment:
Could this variable be moved to outer scope?

Reply:
ok